src/realm/util/file.cpp

```diff
@@ -1098,7 +1098,10 @@ static void _unlock(int m_fd)
     do {
         r = flock(m_fd, LOCK_UN);
     } while (r != 0 && errno == EINTR);
-    REALM_ASSERT_RELEASE_EX(r == 0 && "File::unlock()", r, errno);
+    if (r && errno) {
+        throw RuntimeError(ErrorCodes::RuntimeError,
+                           "File::unlock() has failed, this is likely due to some limitation in the OS file system.");
```

Comment:
Could we create a custom error message for Android. I assume we could do this through the build ifdefs we have available. Something along the lines of:

```
File::unlock() has failed, this is likely due to some limitation in the OS file system. The most common cause is trying to use Realm on external storage. From Android API 30 this is no longer supported. See https://developer.android.com/about/versions/11/privacy/storage for more details.
```

Reply:
Or more specifically just the part of the message about potential problem with external storage. We already use REALM_ANDROID define in a few places there.